### PR TITLE
add ci check

### PR DESCRIPTION
**Pull request by fengshunli**
Signed-off-by: fsl <1171313930@qq.com>

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -297,3 +297,39 @@ jobs:
           fetch-depth: 0
 
       - run: make image
+
+
+  scan_vulnerabilities:
```
Comment (fisherxu) [inline .github/workflows/main.yaml:302]: It's better to add  a new yaml for this? Will be more clear. @fengshunli @vincentgoat

Comment (vincentgoat) [reply .github/workflows/main.yaml:302]: Agree. It's better to add a new yaml in the same dir and move this new job `scan_vulnerabilities` into it. @fengshunli

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -297,3 +297,39 @@ jobs:
...
+
+      - name: Run trivy on git repository
+        run: |
+          trivy fs --format table --ignore-unfixed --skip-dirs website --security-checks vuln .
+
+      - name: Build docker images
+        run: make image
+
+      - name: Run trivy on images
+        run: |
```
Comment (vincentgoat) [inline .github/workflows/main.yaml:329]: There is an official github action that can be used to scan images. https://github.com/aquasecurity/trivy-action

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -297,3 +297,39 @@ jobs:
...
+          images=`docker images | grep kubeedge | awk '{print $1":"$2}'`
+          for img in ${images[@]}; do
```
Comment (vincentgoat) [inline .github/workflows/main.yaml:331]: Please use the `strategy.matrix` to run the jobs concurrently.
FYI: https://github.com/kubeedge/kubeedge/blob/master/.github/workflows/release.yml#L15

File: .github/workflows/main.yaml [hunk 4]
```diff
@@ -296,4 +296,4 @@ jobs:
         with:
           fetch-depth: 0
 
-      - run: make image
+      - run: make image
```
Comment (vincentgoat) [inline .github/workflows/main.yaml:299]: This file should not be updated.

File: .github/workflows/scan.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+          pushd $(mktemp -d)
+          wget https://github.com/aquasecurity/trivy/releases/download/v${{ env.TRIVY_VERSION }}/trivy_${{ env.TRIVY_VERSION }}_Linux-64bit.tar.gz
+          tar zxvf trivy_${{ env.TRIVY_VERSION }}_Linux-64bit.tar.gz
+          echo "$(pwd)" >> $GITHUB_PATH
+        env:
+          TRIVY_VERSION: "0.32.1"
+
+      - name: Run trivy on git repository
+        run: |
+          trivy fs --format table --ignore-unfixed --skip-dirs website --security-checks vuln .
```
Comment (vincentgoat) [inline .github/workflows/scan.yml:41]: > --skip-dirs website

The kubeedge repo doesn't contain the dir `website`. Do you mean the dir `Docs`?

File: .github/workflows/scan.yml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+          - edgesite-server
+          - csidriver
+          - iptablesmanager
+          - edgemark
+          - installation-package
+          - controllermanager
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c
+
```
Comment (vincentgoat) [inline .github/workflows/scan.yml:29]: Steps need to be added as follows:
```
      - name: install QEMU
        uses: docker/setup-qemu-action@v1
      - name: install Buildx
        uses: docker/setup-buildx-action@v1
```

File: .github/workflows/scan.yml [hunk 7]
```diff
@@ -0,0 +1,51 @@
+name:  scan vulnerabilities
```
Comment (vincentgoat) [inline .github/workflows/scan.yml:1]:
```suggestion
name: Scan Vulnerabilities
```

Comment (fengshunli) [reply .github/workflows/scan.yml]: > Scan Vulnerabilities

@vincentgoat updated

File: .github/workflows/scan.yml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+      - name: checkout code
+        uses: actions/checkout@v3
+
+      - name: install QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: install Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build docker images
+        run: make crossbuildimage WHAT=${{ matrix.target }}
```
Comment (vincentgoat) [inline .github/workflows/scan.yml:36]:
```suggestion
        run: make image WHAT=${{ matrix.target }}
```

**Pull request closed without merging**